specs/waku/waku-0.md

```diff
@@ -0,0 +1,600 @@
+
+## Definitions
+
+| Term            | Definition                                            |
+| --------------- | ----------------------------------------------------- |
+| **Light node**  | A Waku node that does not forward any messages.       |
+| **Envelope**    | Messages sent and received by Waku nodes.             |
+| **Node**        | Some process that is able to communicate for Waku.    |
+
+## Underlying Transports and Prerequisites
+
+### Use of DevP2P
+
+For nodes to communicate, they MUST implement devp2p and run RLPx. They MUST have some way of connecting to other nodes. Node discovery is largely out of scope for this spec, but see the appendix for some suggestions on how to do this.
+
+### Gossip based routing
+
+In Whisper, messages are gossiped between peers. Whisper is a form of rumor-mongering protocol that works by flooding to its connected peers based on some factors. Messages are eligible for retransmission until their TTL expires. A node SHOULD relay messages to all connected nodes if an envelope matches their PoW and bloom filter settings. If a node works in light mode, it MAY choose not to forward envelopes. A node MUST NOT send expired envelopes, unless the envelopes are sent as a [mailserver](./mailserver.md) response. A node SHOULD NOT send a message to a peer that it has already sent before.
+
+## Wire Specification
+
+### Use of RLPx transport protocol
+
+All Waku messages are sent as devp2p RLPx transport protocol, version 5[^1] packets. These packets MUST be RLP-encoded arrays of data containing two objects: packet code followed by another object (whose type depends on the packet code).  See [informal RLP spec](https://github.com/ethereum/wiki/wiki/RLP) and the [Ethereum Yellow Paper, appendix B](https://ethereum.github.io/yellowpaper/paper.pdf) for more details on RLP.
+
+Waku is a RLPx subprotocol called `waku` with version `0`. The version number corresponds to the major version in the header spec. Minor versions should not break compatiblity of `waku`, this would result in a new major. (Some expections to this apply in the Draft stage of where client implementation is rapidly change).
+
+### ABNF specification
+
+Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) we have the following format:
+
+```abnf
+; Packet codes 0 - 127 are reserved for Waku protocol
+packet-code = 1*3DIGIT
+
+; rate limits
+limit-ip     = 1*DIGIT
+limit-peerid = 1*DIGIT
+limit-topic  = 1*DIGIT
+
+rate-limits = "[" limit-ip limit-peerid limit-topic "]"
+
+pow-requirement-key = 48
```

Comment:
Why must these be changed? Can't it be worked around in status-go temporarily?

Reply:
Ah woops, this is the old specification. I was confused with comment lower down.

Still, old specification had the correct values. Why mess this up now? Just because status-go was not correct?